core/src/main/scala/org/bitcoins/core/script/ScriptOperationFactory.scala

```diff
@@ -52,8 +52,17 @@ trait ScriptOperationFactory[T <: ScriptOperation] extends BitcoinSLogger {
   }
 
   /** Finds a [[org.bitcoins.core.script.ScriptOperation ScriptOperation]] from a given [[scala.Byte Byte]]. */
-  def fromByte(byte: Byte): T = {
-    operations.find(_.toByte == byte).get
+  @inline final def fromByte(byte: Byte): T = {
+    var idx = 0
+    while (idx < operations.length) {
+      val op = operations(idx)
+      if (op.toByte == byte) {
+        return op
+      } else {
+        idx += 1
+      }
+    }
+    sys.error(s"Could not find opcode for byte=${byte}")
```

Comment:
I think a better way to optimize this would be to have a `Map[Byte, ScriptOperation]` so then we can just do `map.get(byte)`

Reply:
so this is actually an interesting question, this has the overhead of a hash function rather than just iterating through a Vector which modern CPUs are really good at doing. I would suspect the `map.get()` operation is slower although I have no hard data for this.

Reply:
It's pretty clear from profiling locally the Map solution is superior
![Screenshot from 2020-05-18 15-21-52](https://user-images.githubusercontent.com/3514957/82274184-1c9bff00-9945-11ea-8f2d-4fcea7cc8195.png)